crates/CrateDeployer.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2021 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.google.api.client.http.*
+import com.google.api.client.http.javanet.NetHttpTransport
+import picocli.CommandLine
+import picocli.CommandLine.Command
+import picocli.CommandLine.Option
+import picocli.CommandLine.Parameters
+import java.io.File
+import java.util.concurrent.Callable
+import kotlin.system.exitProcess
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+
+
+enum class CrateRepoType {
+    Snapshot,
+    Release,
+}
+
+fun httpPut(url: String, token: String, content: ByteArray): HttpResponse {
```

Comment:
let's make it a `private fun` since it's only used in this class right?

Reply:
@lolski addressed in https://github.com/vaticle/bazel-distribution/pull/322/commits/4a3869852e024c99f6e4921b0bcf00a031402f9b